src/clr-angular/utils/focus/key-focus/key-focus.spec.ts

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ClrKeyFocus } from './key-focus';
+import { Component, DebugElement, ViewChild, ElementRef, OnInit } from '@angular/core';
+import { ClrKeyFocusModule } from './key-focus.module';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+@Component({
+  template: `
+    <button class="btn" #enter>Enter focus</button>
+    <div *ngIf="open" [clrKeyFocus]="direction" [clrFocusOnLoad]="focusOnLoad">
+      <button class="btn">Button 1</button>
+      <button class="btn">Button 2</button>
+      <button class="btn">Button 3</button>
+    </div>
+    <button class="btn">Exit focus</button>
+  `,
+})
+class TestComponent implements OnInit {
+  @ViewChild('enter', { static: true, read: ElementRef })
+  enterElement: ElementRef;
+  @ViewChild(ClrKeyFocus, { static: true })
+  keyFocus: ClrKeyFocus;
+
+  open: boolean = false;
+  direction: string = 'vertical';
+  focusOnLoad = true;
+
+  ngOnInit() {
+    this.enterElement.nativeElement.focus();
+  }
+}
+
+let fixture: ComponentFixture<any>;
+let component: TestComponent;
+
+let directiveDebugElement: DebugElement;
+let clarityDirective: ClrKeyFocus;
+let clarityElement: HTMLElement;
+
+const openMenu = () => {
+  component.open = true;
+  fixture.detectChanges();
+
+  directiveDebugElement = fixture.debugElement.query(By.directive(ClrKeyFocus));
+  clarityElement = directiveDebugElement.nativeElement;
+  clarityDirective = directiveDebugElement.injector.get(ClrKeyFocus);
+};
+
+const keyPress = (key: string) => {
+  clarityElement.dispatchEvent(new KeyboardEvent('keydown', { key: key }));
+};
+
+describe('KeyFocus directive', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [ClrKeyFocusModule],
+      declarations: [TestComponent],
+    });
+    fixture = TestBed.createComponent(TestComponent);
+    fixture.detectChanges();
+
+    component = fixture.componentInstance;
+  });
+
+  describe('Navigation', () => {
+    it('focus on load', () => {
+      expect(document.activeElement.textContent).toBe('Enter focus');
```

Comment:
I don't think we need this test. It is only testing the setup done in the test component initialization. It doesnt seem to test any part of the key-focus directive.

Reply:
I thought it would be cleaner if we proved that focus was somewhere else before opening the menu. Though it may be implied, as the content is ngIf-ed at that 1st moment...
Ok, will remove it.

Reply:
done